### PR TITLE
Feature/edit profile be

### PR DESCRIPTION
**Pull request by zennarchi**

### CODE REVIEW

File: client/src/components/EmailChange/index.tsx [hunk 1]
```diff
@@ -2,10 +2,12 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { changeEmail, setEmailText } from '../../store/emailChange/actions';
 import {Button, Form, Input} from 'reactstrap';
+import {IUser} from '../../store/commonInterfaces/commonInterfaces';
```
Comment (InnaKutsik) [inline client/src/components/EmailChange/index.tsx]: spaces

File: client/src/store/emailChange/actions.ts [hunk 2]
```diff
@@ -1,20 +1,21 @@
 import { SET_EMAIL_TEXT } from './actionTypes';
 import { Dispatch } from 'redux';
-import { store } from '../../App';
 import { showAlert } from '../alert/actions';
+import {AuthAPI} from '../api';
+import {IUser} from '../commonInterfaces/commonInterfaces';
```
Comment (InnaKutsik) [inline client/src/store/emailChange/actions.ts]: spaces

File: server/src/controllers/profile.controllers/changeEmail.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {NextFunction, Request, Response} from 'express';
+import validateInput, {IValidationError} from '../../helpers/validation';
+import {sendChangingEmail} from '../../helpers/send.email';
+import {changeEmailMessage} from '../../helpers/send.email.change.email';
+import Validator from 'validator';
+
+export const changeEmail = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+        const { newEmail, profileUser: {email} }: {newEmail: string, profileUser: {email: string}} = req.body;
```
Comment (InnaKutsik) [inline server/src/controllers/profile.controllers/changeEmail.ts]: pls create interface for {newEmail: string, profileUser: {email: string}}
it is hard to read

File: server/src/controllers/profile.controllers/confirm.change.email.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import {NextFunction, Request, Response} from 'express';
+import {decodeJWT} from '../../helpers/jwt.encoders';
+import {IUserModel, User} from '../../models/user.model';
+
+export const confirmChangeEmail = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+        const { oldEmail, email, token }: {oldEmail: string, email: string, token: string} = req.params;
```
Comment (InnaKutsik) [inline server/src/controllers/profile.controllers/confirm.change.email.ts]: pls create interface

File: server/src/db.requests/user.requests.ts [hunk 5]
```diff
@@ -60,3 +60,29 @@ export const checkUserByProp = async (prop: string, done: any): Promise<IUserMod
...
     }
 };
+
+export const editUser = async (
+    userEmail: string,
+    newUser: any,
+    next: NextFunction,
+): Promise<IUserModel | void | null> => {
+    try {
+        const { username, fullName, description }: any = newUser;
```
Comment (InnaKutsik) [inline server/src/db.requests/user.requests.ts]: any?

File: client/src/components/EmailChange/index.tsx (hunk 1 above)
Comment (zennarchi) [reply client/src/components/EmailChange/index.tsx]: Fixed

File: client/src/store/emailChange/actions.ts (hunk 2 above)
Comment (zennarchi) [reply client/src/store/emailChange/actions.ts]: Fixed

File: server/src/controllers/profile.controllers/changeEmail.ts (hunk 3 above)
Comment (zennarchi) [reply server/src/controllers/profile.controllers/changeEmail.ts]: Fixed

File: server/src/controllers/profile.controllers/confirm.change.email.ts (hunk 4 above)
Comment (zennarchi) [reply server/src/controllers/profile.controllers/confirm.change.email.ts]: Fixed

File: server/src/db.requests/user.requests.ts (hunk 5 above)
Comment (zennarchi) [reply server/src/db.requests/user.requests.ts]: Fixed

**Pull request merged**